### PR TITLE
Fix `PrestashopLogger`  attr `object_type` validate

### LINKED ISSUE #28617: Property `PrestaShopLogger->object_type` is not valid
### Prerequisites

- [X] I understand and accept the project's [code of conduct](https://github.com/PrestaShop/PrestaShop/blob/develop/CODE_OF_CONDUCT.md)
- [X] I have already [searched in existing issues](https://github.com/PrestaShop/PrestaShop/issues?q=is%3Aissue+label%3ABug) and found no previous report of this bug.

### Describe the bug and add screenshots

Property `PrestaShopLogger->object_type` is not valid when object class name contains `_`

image

### Expected behavior

Validate object class name according PHP doc https://www.php.net/manual/en/language.oop5.basic.php

### Steps to reproduce

Try this code 

```php
<?php

PrestaShopLogger::addLog(
     'My log message'
      1,
      null,
      'My_ClassName',
      666,
      true,
      1
 );
```

The log will not be saved and you will get an error. This is because the current `PrestaShopLogger->object_type` validate definition is `isName` so `My_ClassName` is not valid.

NOTE :  This logger is used in the admin controllers to record actions performed on model objects. So all controllers having a className with an underscore will display an error message when add objects.

```php
<?php

class AdminPbsShortCodeController extends ModuleAdminController {
    public function __construct()
    {
        $this->bootstrap = true;
        $this->table = 'my_table_name';
        $this->identifier = 'id_my_table_name';
        $this->className = 'My_ClassName';
        $this->lang = true;

        parent::__construct();
    }

}
```

See for example `AdminController::processAdd`, the ` $this->className` attr is used to log add action, and it will be considered invalid because is not name, according the `PrestaShopLogger` definition.

### PrestaShop version(s) where the bug happened

1.7.8.6

### PHP version(s) where the bug happened

_No response_

### If your bug is related to a module, specify its name and its version

_No response_

### PR DESCRIPTION
**Pull request by PululuK**
| Questions         | Answers
| ----------------- | -------------------------------------------------------
| Branch?           | develop
| Description?      | Fix `PrestashopLogger`  attr `object_type` validate. Validate object class name according PHP doc https://www.php.net/manual/en/language.oop5.basic.php
| Type?             | bug fix 
| Category?         | CO
| BC breaks?        | no
| Deprecations?     | no
| Fixed ticket?     | Fixes https://github.com/PrestaShop/PrestaShop/issues/28617
| Related PRs       | N/A.
| How to test?      | See https://github.com/PrestaShop/PrestaShop/issues/28617.
| Possible impacts? | N/A

### CODE REVIEW

File: classes/Validate.php [hunk 1]
```diff
@@ -1286,4 +1287,16 @@ public static function isValidImapUrl($imapUrl)
 
         return true;
     }
+
+    /**
+     * Check the given string is a valid object class name
```
Comment (matks) [inline classes/Validate.php]:
```suggestion
     * Check the given string is a valid PHP class name
```
In order to remove any ambiguity ? 😉

File: tests/Unit/Adapter/ValidateTest.php [hunk 2]
```diff
@@ -124,4 +124,31 @@ public function isUnsignedIntProvider(): array
...
+            [true, 'MyClassName'],
+            [true, '_MyClassName'],
+            [true, '_My_Class_Name_'],
+            [true, '_MyClassName_'],
+            [true, '__My__Class__Name__'],
+            [false, ''],
+            [false, '666'],
+            [true, '_666'],
+            [true, '_6_6_6_'],
+            [true, '__'],
```
Comment (matks) [inline tests/Unit/Adapter/ValidateTest.php:151]: 😅 not sure this is a good usecase

**Review by matks — COMMENTED** [review]
Good contribution @PululuK I provided some improvement suggestions feel free to add or dismiss

Nice-to-have question: do we really need to have a validation on field `object_type`? 

I think people before added `isName` to avoid random stuff to be put there, but actually I'm not sure we should prevent anything to be put here. A simple `notBlank` could be enough... what is your opinion?

**Comment by PululuK** [comment]
> Good contribution @PululuK I provided some improvement suggestions feel free to add or dismiss
> 
> Nice-to-have question: do we really need to have a validation on field `object_type`?
> 
> I think people before added `isName` to avoid random stuff to be put there, but actually I'm not sure we should prevent anything to be put here. A simple `notBlank` could be enough... what is your opinion?

Data consistency ?

File: tests/Unit/Adapter/ValidateTest.php (hunk 2 above)
Comment (PululuK) [reply tests/Unit/Adapter/ValidateTest.php:151]: @matks can you please provide some good use cases for example ? Or is just a [Hexakosioihexekontahexaphobia](https://www.verywellmind.com/hexakosioihexekontahexaphobia-2671858) ?

**Review by Progi1984 — APPROVED** [review]

File: tests/Unit/Adapter/ValidateTest.php (hunk 2 above)
Comment (matks) [reply tests/Unit/Adapter/ValidateTest.php:151]: Do we consider `__` a valid object type? 🤔

Comment (Progi1984) [reply tests/Unit/Adapter/ValidateTest.php:151]: Yes `__` is a valid object class name.

Example :  
image
https://onlinephp.io?s=s7EvyCjg5QLB5JzE4mKF-HiFal4uTiAqKE3KyUxWKC5JLAFSaaV5ySWZ-XkKGak5OfkammBVnGWJRfEppbkFGuoeIGF1TWugaC0vVy3IwPh4KyuoamsA&v=7.3.33

**Review by matks — APPROVED** [review]
OK for me. The current solution is suitable 👍

**Comment by sLorenzini** [comment]
hi @PululuK,

This PR is too technical for me to test.
Ping @PrestaShop/prestashop-core-developers, can someone try to test this PR 🙏?

Thank you

**Comment by kpodemski** [comment]
Thank you, @PululuK - it works as expected, and there should be no issue with BC Break as the validation should be valid for the previous method + allow usage of other proper names like `My_ClassName`, etc. I've tested provided example and it works as expected:
https://capture.dropbox.com/ZedebWn59DaLdGrC

**Pull request merged**